### PR TITLE
changes for macOS Startup

### PR DESCRIPTION
**Pull request by rwilkey**
These changes are proposed as solution to oetiker/znapzend issue #356 and maybe issue #322.

1. Change znapzend startup behavior to die if no zfs snap or backup sets are found.
2. Update init directory to add a launchd template and modify README.md to explain use.
3. Update main README.md to mention XCode command-line tools requirement.

### CODE REVIEW

**Comment by rwilkey** [comment]
Ok. From what I see in the CI results. it looks like adding the org.oetiker.znapzend.plist.in file for launchd caused problems.  I did not modify any config or makefiles when I built this locally, as this is beyond my capability and probably authority!

File: lib/ZnapZend.pm [hunk 1]
```diff
@@ -129,7 +129,7 @@ my $refreshBackupPlans = sub {
     $self->backupSets($self->zConfig->getBackupSetEnabled($dataSet));
 
     @{$self->backupSets}
-        or $self->zLog->warn("No backup set defined or enabled, yet. run 'znapzendzetup' to setup znapzend");
+        or die "No backup set defined or enabled, yet. run 'znapzendzetup' to setup znapzend";
```
Comment (oetiker) [inline lib/ZnapZend.pm]: can you add a .. \n" at the end of your string so that the error message looks nice in the output.

**Comment by oetiker** [comment]
@rwilkey your launchd.plist.in was fine ... no need to remove it ... I like it much better than large code blobs in the readme ... the reason for the test failure is that you are introducing a new error condition which used to be a warning condition ... you have to update the tests accordingly ... see https://travis-ci.org/oetiker/znapzend/jobs/377530186#L1232

File: lib/ZnapZend.pm (hunk 1 above)
Comment (rwilkey) [reply lib/ZnapZend.pm]: I think I did.

**Comment by oetiker** [comment]
the entry in configure.ac is now missing from your patch ...

also you can call the file org.znapzend since we own this domain.

**Comment by rwilkey** [comment]
@oetiker I see your comments. I assume from most recent, that you want the plist file to be named 'org.znapzend.plist.in' ?
also I see the test error in the log, but am unsure how to correct. I tried changing the test to look for 255 return but that did not do it (or the change never made it in).
I will look at the autoconf.ac

it appears merging is blocked so how do I proceed?

**Comment by oetiker** [comment]
well for one you have to add the plist.in file to configure.ac so that a plist (without the .in) gets generated.

**Comment by rwilkey** [comment]
@oetiker looks like this may be cleaned up. tests are passing and expecting the 'die' condition. Please review when you have time.

Thanks

**Comment by oetiker** [comment]
thanks

**Comment by rwilkey** [comment]
thanks for all the help in bearing with me and getting this done.

**Pull request merged**